Docs about custom claims in MC gateway forwarded requests (#2783)
commercetools/merchant-center-application-kit: website/src/content/api-reference/commercetools-frontend-application-shell.mdx

    context: createApolloContextForProxyForwardTo({
      // The URL to your external API
      uri: externalApiUrl,
      // Provide custom HTTP headers (optional)
      headers: {
        'x-foo': 'bar',
      },
      // Define the audience policy (optional)
      audiencePolicy: 'forward-url-full-path',

>       // Ask for users permissions to be included in exchange JWT (optional)

Reviewer: ```suggestion
      // Set `"X-Forward-To-Claims": "permissions"` header in the request (optional).
```

Author: @Anshuman71 if we include the header name in the comment, I'm wondering if we should also change the comment in the line above about the `audiencePolicy` property (which includes the `X-Forward-To-Audience-Policy` header in the request) to have a consistent messaging.
Otherwise, we would have something like this:
```
// Optionally define the audience policy
audiencePolicy: 'forward-url-full-path',
// Set `"X-Forward-To-Claims": "permissions"` header in the request (optional)
includeUserPermissions: true
```

Should we use that format or should we make both comments similar?
```
// Set `"X-Forward-To-Audience-Policy"` header in the request with provided value (optional)
audiencePolicy: 'forward-url-full-path',
// Set `"X-Forward-To-Claims": "permissions"` header in the request (optional)
includeUserPermissions: true
```

Reviewer: That sounds good. This is a good practice when the options are explained in the docs below. The comments should only say what happened in that line and readers should fall back to the docs for more information.

Author: Updated in this commit: c8053312999955e37e23bc1c73a9fb8896af5314

      },
      // Optionally define the audience policy
      audiencePolicy: 'forward-url-full-path',

>       // Optionally ask for users permissions to be included in exchange JWT

Reviewer: ```suggestion
      // Include user permissions in the JWT (optional)
```

Reviewer: @CarlosCortizasCT, this applies to all such comments.

Author: I partially updated it in this commit: 9d0903a0fd069c9b485bb1bde48621e7d29fddd1

However, I believe the new sentence suggests the JWT is part of what the `createApolloContextForProxyForwardTo` function creates and that's not the case. This new property will make the result of this function call to contain the `X-Forward-To-Claims` header with the `permissions` value so the MC Gateway will include user's permission in the exchange JWT that will get sent along the request to the external service. 
That's why I think we might want to keep the reference to the exchange token. Perhaps we need to find a better way to explain it.

Author: Updated in this commit: c8053312999955e37e23bc1c73a9fb8896af5314

* `uri` (**required**): The URL of the external API to forward the request to.
* `headers` (optional): Additional HTTP headers to be included in the request to the external API.
* `audiencePolicy` (optional): See [configuring the audience policy](/concepts/integrate-with-your-own-api#configuring-the-audience-policy).

> * `includeUserPermissions` (optional): Whether to include an additional token claim in forwarded request containing the logged-in user permissions. See [configuring custom claims](/concepts/integrate-with-your-own-api#configuring-custom-claims)

Reviewer 2: ```suggestion
* `includeUserPermissions` (optional): See [configuring custom claims](/concepts/integrate-with-your-own-api#configuring-custom-claims).
```

Author: Updated in this commit: 088fb79b95c5c8d8d309617db4f8f9f1a47da13b

  * `uri` (**required**): The URL of the external API to forward the request to.
  * `headers` (optional): Additional HTTP headers to be included in the request to the external API.
  * `audiencePolicy` (optional): See [configuring the audience policy](/concepts/integrate-with-your-own-api#configuring-the-audience-policy).

>   * `includeUserPermissions` (optional): Whether to include an additional token claim in forwarded request containing the logged-in user permissions. See [configuring custom claims](/concepts/integrate-with-your-own-api#configuring-custom-claims)

Reviewer 2: I would simply point to the other docs section, where all the necessary info are defined. No need to duplicate part of it.

```suggestion
  * `includeUserPermissions` (optional): See [configuring custom claims](/concepts/integrate-with-your-own-api#configuring-custom-claims).
```

Author: Updated in this commit: 088fb79b95c5c8d8d309617db4f8f9f1a47da13b
